cluster/kubernetes/manifests.go

```diff
@@ -7,12 +7,36 @@ import (
 	"github.com/weaveworks/flux/resource"
 )
 
+// namespacer assigns namespaces to manifests that need it (or "" if
+// the manifest should not have a namespace.
+type namespacer interface {
+	correctNamespace(kresource.KubeManifest) error
+}
+
+// Manifests is an implementation Manifests particular to
+// Kubernetes. Aside from loading manifests from files, it does some
+// "post-processsing" to make sure the view of the manifests is what
+// would be applied; in particular, it fills in the namespace of
+// manifests that would be given a default namespace when applied.
 type Manifests struct {
-	FallbackNamespace string
+	Namespacer namespacer
```

Comment:
I find it a bit odd that the field and the constructor are public, but the type isn't.

Reply:
The interface type describes the requirement; for that purpose, it doesn't need to be referred to outside the package. (The compiler will tell you if it turns out what you're putting in that field doesn't meet the requirement).